core/services/ocr2/delegate.go

```diff
@@ -1692,6 +1695,18 @@ func (d *Delegate) newServicesOCR2Functions(
 	return append([]job.ServiceCtx{functionsProvider, thresholdProvider, s4Provider}, functionsServices...), nil
 }
 
+func (d *Delegate) relayConfigsWithDefaults(relayConfig job.JSONConfig) job.JSONConfig {
+	_, ok := relayConfig["defaultTransactionQueueDepth"]
+	if !ok {
+		relayConfig["defaultTransactionQueueDepth"] = d.cfg.OCR2().DefaultTransactionQueueDepth()
+	}
+	_, ok = relayConfig["simulateTransactions"]
+	if !ok {
+		relayConfig["simulateTransactions"] = d.cfg.OCR2().SimulateTransactions()
+	}
+	return relayConfig
```

Comment:
There is no need to return the `RelayConfig` since it is a map:
```suggestion
func (d *Delegate) relayConfigsWithDefaults(relayConfig job.JSONConfig) {
	_, ok := relayConfig["defaultTransactionQueueDepth"]
	if !ok {
		relayConfig["defaultTransactionQueueDepth"] = d.cfg.OCR2().DefaultTransactionQueueDepth()
	}
	_, ok = relayConfig["simulateTransactions"]
	if !ok {
		relayConfig["simulateTransactions"] = d.cfg.OCR2().SimulateTransactions()
	}
```
Or maybe it would be more straightforward to flip this around:
```go
func (relayConfig RelayConfig) applyDefaultsOCR2(cfg ocr2Config) {
	_, ok := relayConfig["defaultTransactionQueueDepth"]
	if !ok {
		relayConfig["defaultTransactionQueueDepth"] = cfg.DefaultTransactionQueueDepth()
	}
	_, ok = relayConfig["simulateTransactions"]
	if !ok {
		relayConfig["simulateTransactions"] = cfg.SimulateTransactions()
	}
}
```
Then call `relayConfig.applyDefaultsOCR2(cfg.OCR2())` :shrug: 
I don't love the name, but otherwise this seems a little simpler.

Reply:
Oh, I didn't realize this was duplicated in the bootstrap delegate. In that case, moving the method to `RelayConfig` makes even more sense. Let's do that.

Reply:
This was somewhat intended as I'm not a big fan of calling a method that doesn't return anything and just mutates the object passed inside.

Reply:
@jmank88 the field RelayConfig that gets accessed here is of type []byte, which is different than the `RelayConfig` struct you mention in the change, so I can either keep it as it is or add the method to `JSONConfig` instead. I can potentially add this to RelayConfig ([]bytes) but that means I'd have to update every call on every relay, which kinda defeats the purpose of having it in one place.

Reply:
I meant `job.JSONConfig`